src/transaction/transaction.ts

```diff
@@ -421,22 +421,32 @@ export async function populateAppCallResources(atc: algosdk.AtomicTransactionCom
     }
 
     if (type === 'account') {
-      txns[txnIndex].txn.appAccounts?.push(algosdk.decodeAddress(reference as string))
+      txns[txnIndex].txn.appAccounts = [...(txns[txnIndex].txn.appAccounts ?? []), algosdk.decodeAddress(reference as string)]
     } else if (type === 'app') {
-      txns[txnIndex].txn.appForeignApps?.push(Number(reference))
+      txns[txnIndex].txn.appForeignApps = [...(txns[txnIndex].txn.appForeignApps ?? []), Number(reference)]
     } else if (type === 'box') {
       const { app, name } = reference as algosdk.modelsv2.BoxReference
-      txns[txnIndex].txn.boxes?.push({ appIndex: Number(app), name: name })
+      txns[txnIndex].txn.boxes = [...(txns[txnIndex].txn.boxes ?? []), { appIndex: Number(app), name }]
+
+      if (app !== 0) {
```

Comment:
This condition won't work for BigInt(0).

Reply:
Ah good catch. Used toString in [41bbfbe](https://github.com/algorandfoundation/algokit-utils-ts/pull/278/commits/41bbfbe727b1cabe53a45209b789985c5b9692c7)